### DISCUSSION TITLE
[AsyncCommand] and [AsyncTargetRpc] for returning values from Rpcs

### DISCUSSION THREAD

**Discussion by Monsoonexe** [FirstGearGames/FishNet#612] [Feature Request]
Hello there, the project I'm working on uses a version of UNet that has been heavily modified to suit our own needs. We found it useful to return values from Rpcs and Cmds, so I added async support to the Weaver and new attributes `[AsyncCommand]` and `[AsyncTargetRpc]` to support code like `string result = await CmdGetString();`.

I understand that FishNet is NOT UNet, but I do understand it uses the same strategy of performing IL post-processing based on attributes and conventions.

Is this a feature that is desired in FishNet? I would consider championing the feature, if it was something the community and devs wanted.

**Comment by FirstGearGames** [comment]
It's something we wouldn't say no to, but am doubtful it will be an easy copy and paste given its IL related.

**Comment by Monsoonexe** [comment]
Oh, it would definitely take more effort than copy-paste. And of course it
would have to hold up to FishNet's testing and quality standards.

I'd want to discuss limitations, conventions, and performance
considerations beforehand, if this is something you're interested in.

(Edit: removed auto-gen email reply quote)

**Comment by FirstGearGames** [comment]
I cannot dedicate any time to this currently and it may be many weeks or months even before I could. It would be unrealistic for me to give you a timeline so I'm going to mark this as declined.

Not to say that the idea isn't something that's been considered, and not something we're against, but business prevents it from being implemented.
